query/physicalplan/aggregate.go

```diff
@@ -443,62 +451,116 @@ func (a *HashAggregate) Callback(ctx context.Context, r arrow.Record) error {
 	return nil
 }
 
-func (a *HashAggregate) Finish(ctx context.Context) error {
-	ctx, span := a.tracer.Start(ctx, "HashAggregate/Finish")
-	defer span.End()
+func (a *HashAggregate) updateGroupByCols(row int, groupByArrays []arrow.Array, groupByFields []arrow.Field) error {
+	// aggregate is the current aggregation
+	aggregate := a.aggregates[len(a.aggregates)-1]
 
-	numCols := len(a.groupByCols) + 1
-	// Each hash that's aggregated by will become one row in the final result.
-	numRows := len(a.hashToAggregate)
+	for i, arr := range groupByArrays {
+		fieldName := groupByFields[i].Name
 
-	groupByFields := make([]arrow.Field, 0, numCols)
-	groupByArrays := make([]arrow.Array, 0, numCols)
-	for _, fieldName := range a.colOrdering {
-		groupByCol, ok := a.groupByCols[fieldName]
-		if !ok {
-			return fmt.Errorf("unknown field name: %s", fieldName)
+		groupByCol, found := aggregate.groupByCols[fieldName]
+		if !found {
+			groupByCol = builder.NewBuilder(a.pool, groupByFields[i].Type)
```

Comment:
This is where I would say we should create a new opt large binary builder (or even just default to arrow's large binary builder to start with). It's essentially just the hash aggregate result that we care about handling this case.